tag/metadata.go

```diff
@@ -0,0 +1,47 @@
+// Copyright 2019, OpenCensus Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package tag
+
+const (
+	// valueTTLNoPropagation prevents tag from propagating.
+	valueTTLNoPropagation = 0
+
+	// valueTTLUnlimitedPropagation allows tag to propagate without any limits on number of hops.
+	valueTTLUnlimitedPropagation = -1
+)
+
+type metadatas struct {
+	ttl int
+}
+
+// Metadata applies metadatas specified by the function.
+type Metadata func(*metadatas)
+
+// WithTTLNoPropagation applies metadata with ttl value of valueTTLNoPropagation.
+// It is predefined for convenience.
+func WithTTLNoPropagation() Metadata {
```

Comment:
What do you think about this?
```go
type TTL struct {
  value int
}

var (
NoPropagation = &TTL{value= 0}
UnlimitedPropagation = &TTL{value= -1}
)

func WithTTL(TTL ttl) Metadata {
	return func(m *metadatas) {
		m.ttl = ttl
	}
}
     

```

Reply:
you could inadvertently modify UnlimitedPropagation and WithTTL would allow any value of TTL. I thought we don't want that at this point

Reply:
@rakyll would you be worried if someone by mistake sets UnlimitedPropagation to nil?

Reply:
Let's do it this way.

Reply:
@bogdandrutu I have already made the change and uploaded. If everything looks fine then can you please approve it?